doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,397 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
+it detects profile data changes and submits the new data to the Kaa cluster as a profile update.
 
+<img src="endpoint-profile-generation/ClientSideEndpointProfileGeneration_0_8_0.png">
 
-## Client-side endpoint profile update ##
-When the endpoint profile information changes as the result of the client operation or user's actions, it is the client implementation responsibility to update the profile via the endpoint SDK API calls. The endpoint SDK detects profile changes by comparing the new profile hash against the previously persisted one. Should there be a change, the endpoint profile management module passes it to the Operations server, which in turn updates the endpoint profile information in the database and revises the endpoint groups membership.
-Client-side endpoint profile example
-The following code block provides a simple client-side endpoint profile schema example.
+### Client-side endpoint profile example ###
 
+For the purpose of this guide we will use a fairly abstract client-side profile schema shown below.
 
 ```json
 {
-    "namespace": "org.myproject",
-    "type": "record",
-    "name": "MyClientSideProfile",
-    "fields": [
-        { "name": "country", "type": "string" },
-        { "name": "city", "type": "string" },
-        { "name": "age", "type": "int" },
+    "type":"record",
+    "name":"Profile",
+    "namespace":"org.kaaproject.kaa.schema.sample.profile",
+    "fields":[
+        {
+            "name":"id",
+            "type":"string"
+        },
         {
-            "name": "hobbies",
-            "type": { "type": "array", "items": "string" }
+            "name":"os",
+            "type":{
+                "type":"enum",
+                "name":"OS",
+                "symbols":[
+                    "Android",
+                    "iOS",
+                    "Linux"
+                ]
+            }
+        },
+        {
+            "name":"os_version",
+            "type":"string"
+        },
+        {
+            "name":"build",
+            "type":"string"
         }
     ]
 }
 ```
 
-## The following client-side profile would be compatible with our schema example. ##
+The following client-side profile would be compatible with our schema example.
 
 ```json
 {
-    "country": "US",
-    "city": "San Francisco",
-    "age": 32,
-    "hobbies": [
-        "skydiving",
-        "hiking"
-    ]
+  "id" : "1",
+  "os" : "Android",
+  "os_version" : "1",
+  "build" : "1"
+}
+```
+The schema structure from our example allows filtering the endpoints by the owner's country (for example, to show only US news), city (for example, to push weather 
+notifications only for the specified cities), age (for example, to apply age restrictions), and hobbies (for example, to push football scores for those whose hobby list 
+contains "football"). It is allowed to create complex filtering conditions by combining as many filtering conditions as needed.
```

Comment:
This text is left from the previous documentation version and does not relate to the actual schema shown above.